src/Illuminate/Filesystem/FilesystemManager.php

```diff
@@ -188,7 +188,7 @@ public function createS3Driver(array $config)
         $root = isset($s3Config['root']) ? $s3Config['root'] : null;
 
         return $this->adapt($this->createFlysystem(
-            new S3Adapter(new S3Client($s3Config), $s3Config['bucket'], $root), $config
+            new S3Adapter(new S3Client($s3Config), $s3Config['bucket'], $root, $config['options'] ?: []), $config
```

Comment:
This will break existing apps if the options index doesn't exist. Please use an isset check.